lightning/src/ln/functional_tests.rs

```diff
@@ -2346,48 +2346,39 @@ fn claim_htlc_outputs_single_tx() {
 		}
 
 		let node_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
-		assert_eq!(node_txn.len(), 21);
+		assert_eq!(node_txn.len(), 9);
 		// ChannelMonitor: justice tx revoked offered htlc, justice tx revoked received htlc, justice tx revoked to_local (3)
 		// ChannelManager: local commmitment + local HTLC-timeout (2)
-		// ChannelMonitor: bumped justice tx (4), after one increase, bumps on HTLC aren't generated not being substantial anymore
-		// ChannelMonito r: local commitment + local HTLC-timeout (14)
-
-		assert_eq!(node_txn[0], node_txn[5]);
-		assert_eq!(node_txn[0], node_txn[7]);
-		assert_eq!(node_txn[0], node_txn[9]);
-		assert_eq!(node_txn[0], node_txn[13]);
-		assert_eq!(node_txn[0], node_txn[15]);
-		assert_eq!(node_txn[0], node_txn[17]);
-		assert_eq!(node_txn[0], node_txn[19]);
-
-		assert_eq!(node_txn[1], node_txn[6]);
-		assert_eq!(node_txn[1], node_txn[8]);
-		assert_eq!(node_txn[1], node_txn[10]);
-		assert_eq!(node_txn[1], node_txn[14]);
-		assert_eq!(node_txn[1], node_txn[16]);
-		assert_eq!(node_txn[1], node_txn[18]);
-		assert_eq!(node_txn[1], node_txn[20]);
-
-
-		// Check the pair local commitment and HTLC-timeout broadcast due to HTLC expiration and present 8 times (rebroadcast at every block from 200 to 206)
-		assert_eq!(node_txn[0].input.len(), 1);
-		check_spends!(node_txn[0], chan_1.3);
-		assert_eq!(node_txn[1].input.len(), 1);
-		let witness_script = node_txn[1].input[0].witness.last().unwrap();
-		assert_eq!(witness_script.len(), OFFERED_HTLC_SCRIPT_WEIGHT); //Spending an offered htlc output
-		check_spends!(node_txn[1], node_txn[0]);
+		// ChannelMonitor: bumped justice tx (2), after one increase, bumps on HTLC aren't generated not being substantial anymore, bump on revoked to_local isn't generated due to more room for expiration
+		// ChannelMonitor: local commitment + local HTLC-timeout (2)
 
-		// Justice transactions are indices 2-3-4
+		// Check the pair local commitment and HTLC-timeout broadcast due to HTLC expiration
 		assert_eq!(node_txn[2].input.len(), 1);
+		check_spends!(node_txn[2], chan_1.3.clone());
 		assert_eq!(node_txn[3].input.len(), 1);
+		let witness_script = node_txn[3].input[0].witness.last().unwrap();
+		assert_eq!(witness_script.len(), OFFERED_HTLC_SCRIPT_WEIGHT); //Spending an offered htlc output
+		check_spends!(node_txn[3], node_txn[2].clone());
```

Comment:
We don't need the clone()s anymore (also true in a number of places - can you do a quick pass and remove all of them?).


Also, while its totally fine at the end, just for future reference (and you do this between some commits here), we can't be broadcasting transactions out of order, so node_txn[higher_index] must always spend node_txn[lower_index].

Reply:
Yes right, tip commit should be fine on this, but was quite hacky to maintain between more. Should be documented in broadcast interface.